src/arch/x86_64.rs

```diff
@@ -73,3 +73,11 @@ fn panic(info: &core::panic::PanicInfo) -> ! {
     vga.set_color(vga::ColorSpec::new(vga::Color::Red, vga::Color::Black));
     mycelium_kernel::handle_panic(&mut vga, info)
 }
+
+#[alloc_error_handler]
+#[cfg(target_os = "none")]
+fn alloc_error(layout: core::alloc::Layout) -> ! {
+    let mut vga = vga::writer();
+    vga.set_color(vga::ColorSpec::new(vga::Color::Red, vga::Color::Black));
+    mycelium_kernel::handle_alloc_error(&mut vga, layout)
+}
```

Comment:
Take it or leave it: this could also just panic, which could be platform-agnostic. Unless we eventually want to do Real Stuff here.

Reply:
True! It can panic as we're not unwinding.
I just put it next to the panic handler 'cause I was cargo-culting everything not directly related to the allocator tbh.